### PR TITLE
ci: fix build-images-ci to not die in forks

### LINKED ISSUE #34947: CI: Image CI Build
## CI failure

> [Image CI Build](https://github.com/check-spelling-sandbox/cilium/actions/runs/10925110001/workflow)
Error when evaluating 'runs-on' for job 'build-and-push-prs'. .github/workflows/build-images-ci.yaml (Line: 39, Col: 14): Unexpected value ''

https://github.com/check-spelling-sandbox/cilium/actions/runs/10925110001

---

The fix is to add an `if: ${{ vars.GH_RUNNER_EXTRA_POWER }}`

### PR DESCRIPTION
**Pull request by jsoref**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

Forks will not necessarily have vars.GH_RUNNER_EXTRA_POWER. The best outcome is to not run a job when the var is not set.

Fixes: #34947

### CODE REVIEW

**Comment by joestringer** [comment]
The change here seems fine, but I think it's worth noting that currently if someone wants to run the automated CI testing for Cilium, it's typically done on a PR towards this repository. Running the CI on forks is not something that the community has yet tried to support. I can see from https://github.com/check-spelling-sandbox/cilium/pull/1 that there are various workflows failing for different reasons. How far are you interested to take this, do you plan to make all the workflows pass or skip on a fork?

This might be a step towards supporting that development workflow, but it's also worth noting that many of the other workflows rely on the images that are built from this workflow, and they won't be able to pass if this step is skipped. Linters should be able to run without this workflow, but the smoke tests and end-to-end tests will time out if this step is skipped.

Also curious if you have been able to test out this change on your own fork to validate the fix?

**Comment by jsoref** [comment]
It works, tested here: https://github.com/jsoref/cilium/actions/runs/10926879711/job/30331778002?pr=1

**Comment by jsoref** [comment]
An alternate approach would be to have the workflow push to ghcr and have the workflows rely on ghcr instead.

**Comment by jsoref** [comment]
To answer the question about how far: my goal is for nothing to give me a ❌. As long as the project is willing to work with me.

File: .github/workflows/build-images-ci.yaml [hunk 1]
```diff
@@ -36,6 +36,7 @@ jobs:
   build-and-push-prs:
     timeout-minutes: 45
     name: Build and Push Images
+    if: ${{ vars.GH_RUNNER_EXTRA_POWER }}
     runs-on: ${{ vars.GH_RUNNER_EXTRA_POWER }}
```
Comment (aanm) [inline .github/workflows/build-images-ci.yaml:40]:
```suggestion
    runs-on: ${{ vars.GH_RUNNER_EXTRA_POWER || 'ubuntu-latest' }}
```

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
This issue has been fixed. Thank you.

**Pull request closed without merging**